lib/tasks/indices.rake

```diff
@@ -156,46 +180,54 @@ You should run this task if the index schema has changed.
     # as we only want to clean the indices for the versions that matches from the overnight copy task
     # if we didn't check this we could potentially attempt to delete one of the new indices as we are importing
     # data into it.
-    index_names.each do |index_name|
-      search_server.index_group(index_name).clean
+    Clusters.active.each do |cluster|
+      index_names.each do |index_name|
+        search_server(cluster: cluster).index_group(index_name).clean
+      end
     end
   end
 
   desc "Check whether a restored index has recovered"
-  task :check_recovery, [:index_name] do |_, args|
+  task :check_recovery, [:index_name, :clusters] do |_, args|
     raise "An 'index_name' must be supplied" unless args.index_name
 
-    puts SearchIndices::Index.index_recovered?(
-      base_uri: elasticsearch_uri,
-      index_name: args.index_name
-    )
+    clusters_from_args(args).each do |cluster|
+      puts "Recovery status of #{args.index_name} on cluster #{cluster.key} (#{cluster.uri}):"
+      puts SearchIndices::Index.index_recovered?(
+        base_uri: cluster.uri,
+        index_name: args.index_name
+      )
+    end
   end
 
   desc "Monitor the search indices and send data to statsd"
   task :monitor_indices do
-    client = Services.elasticsearch
-    statsd = Services.statsd_client
-    missing = []
-
-    SearchConfig.instance.all_index_names.each do |index|
-      begin
-        stats = client.indices.stats index: index, docs: true
-        docs = stats["_all"]["primaries"]["docs"]
-
-        count = docs["count"]
-        statsd.gauge("#{index}_index.docs.count", count)
-        puts "#{index}_index.docs.count=#{count}"
-
-        deleted = docs["deleted"]
-        statsd.gauge("#{index}_index.docs.deleted", deleted)
-        puts "#{index}_index.docs.deleted=#{deleted}"
-
-      rescue Elasticsearch::Transport::Transport::Errors::NotFound
-        missing << index
+    Clusters.active.each do |cluster|
+      client = Services.elasticsearch(cluster: cluster)
+      statsd = Services.statsd_client
+      missing = []
+      cluster_name = "cluster_#{cluster.key}"
+
+      SearchConfig.instance.all_index_names.each do |index|
+        begin
+          stats = client.indices.stats index: index, docs: true
+          docs = stats["_all"]["primaries"]["docs"]
+
+          count = docs["count"]
+          statsd.gauge("#{cluster_name}.#{index}_index.docs.count", count)
```

Comment:
Joining up metrics will be a bit awkward, especially when the B cluster becomes the A cluster, but I'm not sure there's a good way to solve that here.

Reply:
One option would be to have version-like numbers? Rather than clusters having 'A' or 'B' names that signify priority. I figure we could join cluster metrics up with some clever use of `*` in our graphite queries. What do you think?

Reply:
That makes sense, all the infrastructure is already called `elasticsearch5` or `elasticsearch6`.  Though if we ever want to do an A/B test with the same version of elasticsearch (but some other changes to the cluster) then it wouldn't quite fit.

Reply:
Perhaps we could also version the versions?

It's versions all the way down: `elasticsearch5-v1`

Reply:
Let's just keep it simple and use the cluster name for now.

Reply:
Is cluster name (e.g.) `elasticsearch5`?

Reply:
No, the `cluster_A`, `cluster_B`.  We can always come up with a better way later.